lib/rspec/core/configuration.rb

```diff
@@ -230,6 +230,10 @@ def fail_fast=(value)
         end
       end
 
+      # @macro trap_interrupt
+      # Default: `true`.
+      add_setting :trap_interrupt
```

Comment:
I think I'd rather handle this like we handle zero monkey patching mode, that is that theres a method which clears our event handler and restores the previous one.

e.g.
```
def clear_interrupt_handler!
  Signal.trap(:INT, RSpec.world.original_interrupt_handler)
end
```

I don't think we need to conditionally restore ours at any point, it should be a once only deal, you're either going to want this or not.

Reply:
```
      # This is used internally by RSpec to run a suite, but is available
      # for use by any other automation tool.
```

In your own comment, you say other tools may use this entry point.

So, IMHO, it should correctly set and restore interrupt handlers.

> it should be a once only deal, you're either going to want this or not.

What happens if the user calls this method more than once?

What are the invariants regarding signal handling that we want?

Reply:
> What happens if the user calls this method more than once?

I'm ok with it being a no-op, you've called it, we've removed our handlers, you call it again theres nothing for us to do.

Reply:
No, I mean what if the user calls `Runner.run` more than once, maybe with different options?

Reply:
Ah I see, in this case as its a global option it's behaviour would be permanent, (although I'm ok if you want to add a way to undo it too).